fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Transaction_object.java

```diff
@@ -0,0 +1,218 @@
+   * @param map_hash_key Map<String,PublicKey> dictionary public key by public key hash
+   * @return List<PublicKey> outputs public keys
+   */
+  public List<PublicKey> get_receivers_transaction(Map<String, PublicKey> map_hash_key) {
+    Iterator<String> receiver_hash_ite = get_receivers_hash_transaction().iterator();
+    List<PublicKey> receivers = new ArrayList<>();
+    while (receiver_hash_ite.hasNext()){
+      PublicKey pub = map_hash_key.getOrDefault(receiver_hash_ite.next(),null);
+      if (pub == null) {
+        throw new IllegalArgumentException("The hash correspond to no key in the dictionary");
+      }
+      receivers.add(pub);
+    }
+    return receivers;
+  }
+
+  /**
+   * Check if a public key is in the recipient
+   *
+   * @param publicKey PublicKey of someone
+   * @return true if public key in receiver, false otherwise
+   */
+  public boolean is_receiver(PublicKey publicKey) {
+    return get_receivers_hash_transaction().contains(publicKey.computeHash());
+  }
+
+  /**
+   * Function that given a key pair change the sig of an input considering all the outputs
+   *
+   * @param keyPair of one input sender
+   * @return sig other all the outputs and inputs with the public key
+   * @throws GeneralSecurityException
+   */
+  public String compute_sig_outputs_inputs(KeyPair keyPair) throws GeneralSecurityException {
+    // input #1: tx_out_hash Value //input #1: tx_out_index Value
+    // input #2: tx_out_hash Value //input #2: tx_out_index Value ...
+    // TxOut #1: LaoCoin Value​​ //TxOut #1: script.type Value //TxOut #1: script.pubkey_hash Value
+    // TxOut #2: LaoCoin Value​​ //TxOut #2: script.type Value //TxOut #2: script.pubkey_hash
+    // Value...
+    String[] sig = new String[inputs.size() * 2 + outputs.size() * 3];
+    Iterator<Input_object> ite_input = inputs.iterator();
+    Iterator<Output_object> ite_output = outputs.iterator();
+
+    int index = 0;
+    while (ite_input.hasNext()){
+      Input_object current = ite_input.next();
+      sig[index] = current.get_tx_out_hash();
+      sig[index + 1] = String.valueOf(current.get_tx_out_index());
+      index = index + 2;
+    }
+
+    while (ite_output.hasNext()){
+      Output_object current = ite_output.next();
+      sig[index] = String.valueOf(current.get_value());
+      sig[index + 1] = current.get_script().get_type();
+      sig[index + 2] = current.get_script().get_pubkey_hash();
+      index = index + 3;
+    }
+    return keyPair.sign(new Base64URLData(String.join("", sig))).getEncoded();
+  }
+
+  /**
+   * Function that given a Public Key gives the miniLaoCoin received
+   *
+   * @param receiver Public Key of a potential receiver
+   * @return int amount of Lao Coin
+   */
+  public int get_miniLao_per_receiver(PublicKey receiver) {
+    // TODO : Useful check ?
+    // if (!is_receiver(receiver)) {
+    // throw new IllegalArgumentException("The public Key is not contained in the receiver public
+    // key");
+    // }
```

Comment:
I think this should be included it is a good sanity check, but maybe have a more explicit comment